aiohttp/tracing.py

```diff
@@ -24,31 +33,37 @@ class TraceConfig:
     """First-class used to trace requests launched via ClientSession
     objects."""
 
-    def __init__(self, trace_config_ctx_factory=SimpleNamespace):
-        self._on_request_start = Signal(self)
-        self._on_request_chunk_sent = Signal(self)
-        self._on_response_chunk_received = Signal(self)
-        self._on_request_end = Signal(self)
-        self._on_request_exception = Signal(self)
-        self._on_request_redirect = Signal(self)
-        self._on_connection_queued_start = Signal(self)
-        self._on_connection_queued_end = Signal(self)
-        self._on_connection_create_start = Signal(self)
-        self._on_connection_create_end = Signal(self)
-        self._on_connection_reuseconn = Signal(self)
-        self._on_dns_resolvehost_start = Signal(self)
-        self._on_dns_resolvehost_end = Signal(self)
-        self._on_dns_cache_hit = Signal(self)
-        self._on_dns_cache_miss = Signal(self)
-
-        self._trace_config_ctx_factory = trace_config_ctx_factory
-
-    def trace_config_ctx(self, trace_request_ctx=None):
+    def __init__(
+        self,
+        trace_config_ctx_factory: Type[SimpleNamespace]=SimpleNamespace
+    ) -> None:
+        self._on_request_start = Signal(self)  # type: _Signal
```

Comment:
Hm. Do we have any conventions about when to use type comment and when - annotation? Why not to use annotations everywhere? IIRC, those comments were made just for Python 2.x  typing support.

Reply:
I didn't find any documented convention regarding style for annotations so it looks like the question to @asvetlov.

Reply:
1. This comment is for Python 3.5; `a: type = val` is supported by Python 3.6 only.
2. We are in the middle of the transition to a fully annotated code. After the work is done mypy strict mode will be enabled.

The line is correct.

Reply:
I almost forgot what 3.5 is. Thanks!